packages/headless/src/api/search/date/relative-date.ts

```diff
@@ -0,0 +1,161 @@
+  'minute',
+  'hour',
+  'day',
+  'week',
+  'month',
+  'quarter',
+  'year',
+];
+
+/**
+ * Defines a date relative to the current moment.
+ */
+export interface RelativeDate {
+  /**
+   * The relative period of time.
+   */
+  period: RelativeDatePeriod;
+  /**
+   * The unit of time in which the date is set relative to. When `period` is set as `now`, the `unit` does not have to be defined.
+   */
+  unit?: RelativeDateUnit;
+  /**
+   * The amount of the `unit` of time. When `period` is set as `now`, the `amount` does not have to be defined.
+   */
+  amount?: number;
+  /**
+   * If `true`, the date and time will be returned based on local settings . If `false`, the date will be adjusted to _Coordinated Universal Time_ (UTC).
+   *
+   * @defaultValue `false`
+   */
+  useLocalTime?: boolean;
+}
+
+const buildRelativeDateDefinition = (period: RelativeDatePeriod) => {
+  const isNow = period === 'now';
+  return {
+    amount: new NumberValue({required: !isNow, min: 1}),
+    unit: new StringValue({
+      required: !isNow,
+      constrainTo: validRelativeDateUnits,
+    }),
+    period: new StringValue({
+      required: true,
+      constrainTo: validRelativeDatePeriods,
+    }),
+    useLocalTime: new BooleanValue(),
+  };
+};
+
+function validateRelativeDate(relativeDate: RelativeDate) {
+  new Schema(buildRelativeDateDefinition(relativeDate.period)).validate(
+    relativeDate
+  );
+}
+
+/**
+ * Formats a `RelativeDate` object to a parsable string value.
+ * @param relativeDate The `RelativeDate` object.
+ */
+export function formatRelativeDate(relativeDate: RelativeDate) {
+  validateRelativeDate(relativeDate);
+  const {period, amount, unit, useLocalTime} = relativeDate;
+  const utc = useLocalTime ? '' : 'utc';
+
+  switch (period) {
+    case 'past':
+    case 'future':
+      return `${period}${amount}${unit}${utc}`;
+    case 'now':
+      return `${period}${utc}`;
+  }
+}
+
+export function formatRelativeDateForSearchApi(date: string) {
+  const relativeDate = parseRelativeDate(date);
+  const {period, amount, unit, useLocalTime} = relativeDate;
+  const dayjsDate = useLocalTime ? dayjs() : dayjs().utc();
+
+  switch (period) {
+    case 'past':
+      return formatDateForSearchApi(
+        dayjsDate.subtract(amount!, unit as QUnitType)
+      );
+    case 'future':
+      return formatDateForSearchApi(dayjsDate.add(amount!, unit as QUnitType));
+    case 'now':
+      return formatDateForSearchApi(dayjsDate);
+  }
+}
+
+const relativeDateFormatRegexp = new RegExp(
+  `^((?<period>past|future)(?<amount>\\d+)(?<unit>${validRelativeDateUnits.join(
+    '|'
+  )})|(?<now>now))(?<utc>utc)?$`,
+  'i'
+);
+
+export function isRelativeDateFormat(date: string) {
+  if (date.toLowerCase() === 'now') {
+    return true;
+  }
+
+  return relativeDateFormatRegexp.test(date);
+}
+
+export function isRelativeDate(date: unknown): date is RelativeDate {
+  return date && typeof date === 'object' && 'period' in date ? true : false;
```

Comment:
why do we need the `?` operator? Why not just do `return date && typeof date === 'object' && 'period' in date`?

Reply:
The returned type absolutely has to be a boolean in this case, even setting `!!date` in the beginning of the condition did not work

Reply:
Since `&&` returns the first falsy value (or the last value if all are truthy), the return type in your example would be `unknown | false` because of the first condition. Adding a `!!` would keep it boolean:
`return !!date && typeof date === 'object' && 'period' in date;`

Reply:
`return !!date && typeof date === 'object' && 'period' in date;` returns an error
`return !!date && typeof date === 'object' && 'period' in date!;` works

Reply:
Apologies for my redundant message, the PR didn't update on my end and I didn't see your comments yet.